transport/zipkin/http.go

```diff
@@ -160,7 +162,17 @@ func (c *HTTPTransport) send(spans []*zipkincore.Span) error {
 		req.SetBasicAuth(c.httpCredentials.username, c.httpCredentials.password)
 	}
 
-	_, err = c.client.Do(req)
+	resp, err := c.client.Do(req)
+
+	if resp != nil && resp.StatusCode >= http.StatusBadRequest {
+		body := ""
+		if resp.Body != nil {
```

Comment:
to yuri's point
 // The http Client and Transport guarantee that Body is always
 // non-nil, even on responses without a body or responses with
 // a zero-length body.

good to know.